[WIP] Update the CCADB Policy to Version 2.0 (#198)
mozilla/www.ccadb.org: policy.md

A root CA certificate is a self-signed certificate issued by the root CA to identify itself and to facilitate verification of certificates issued by its corresponding PKI hierarchy. 

CA Owners MUST maintain correct and current information about their root CA certificates. If root CA certificate information needs to be added or updated, the CA Owner MUST submit an [Add/Update Root Request Case](https://www.ccadb.org/cas/updates).

### 3.2 Subordinate CA Certificates

A subordinate CA certificate is a certificate signed by a root CA or another subordinate CA. The CCADB considers the terms "intermediate" and "subordinate" synonymous. 

CA Owners MUST disclose:
- all subordinate CA certificates capable of validating to a certificate included in a Root Store or associated with a CCADB Root Inclusion Request. Disclosure MUST take place within 7 calendar days of issuance and before the subject CA represented in the certificate begins issuing publicly-trusted certificates.
- revocation of all subordinate CA certificates capable of validating to a certificate included in a Root Store or associated with a CCADB Root Inclusion Request within 7 calendar days of revocation.

> CA certificates created by cross-signing are considered subordinate CA certificates by Root Store Operators and MUST be disclosed to both the issuer and subject CA Owner PKI hierarchies in the CCADB. 

Reviewer: I wonder to which extent both CA Owners need to disclose this. 

If CA1 has issued SubCAX - issued by CA1, they would have to disclose this SubCA in CCADB.
If CA2 then cross-signs SubCAX as well, current practice seems to be that CA2 needs to disclose SubCAX - issued by CA2.

Does this new language intend to confirm that practice, **or** is the intent here that even CA1 needs to disclose the SubCAX - issued by CA2 certificate within their CCADB account?

Reviewer 2: > If CA1 has issued SubCAX - issued by CA1, they would have to disclose this SubCA in CCADB. 

Yes, under the CA1 hierarchy. This is necessary due to the earlier statement: “_all subordinate CA certificates capable of validating to a certificate included in a Root Store or associated with a CCADB Root Inclusion Request. Disclosure MUST take place within 7 calendar days of issuance and before the subject CA represented in the certificate begins issuing publicly-trusted certificates._”

> If CA2 then cross-signs SubCAX as well, current practice seems to be that CA2 needs to disclose SubCAX - issued by CA2. 

Yes, under the CA2 hierarchy.

> Does this new language intend to confirm that practice, or is the intent here that even CA1 needs to disclose the SubCAX - issued by CA2 certificate within their CCADB account?

This language was intended to confirm the practice you highlighted with your CA1 and CA2 example, but we think we can be more clear with this suggested change:

``` suggestion
Cross-certificates (i.e., where the same subject and public key of an existing CA certificate appears in at least one additional certificate issued by a _different_ CA Owner) are considered subordinate CA certificates by Root Store Operators and MUST be disclosed to the issuing CA Owner’s PKI hierarchy in the CCADB. 

Ultimately, this means the subject CA certificate may appear in two distinct CA Owner PKI hierarchies, but it MUST minimally appear in the issuing CA Owner’s.
```

Reviewer: Thanks! That matches with what I was thinking.

# CCADB Policy

*Version 2.0, Effective: June 15, 2025*

## Introduction

Several Public Root Store Operators have collaborated to create the Common Certification Authority Database (CCADB), a data repository of certificate and Certification Authority (CA) information. 

CA Owners who wish to be included in a CCADB-participating Root Store ("Root Store") need to disclose and maintain certain information in the CCADB. This policy describes common Root Store Operator requirements related to CA Owner use of the CCADB. This policy does not cover how to obtain write access to the CCADB or how to use the CCADB’s web interface. Additional information covering these topics is available [here](https://www.ccadb.org/cas/).

> Root Store Operators MAY have additional CCADB-related requirements defined in their own Root Store policies. These policies MAY strengthen the requirements described in this policy. Additional information on the policies belonging to the individual Root Store Operators participating in the CCADB can be found [here](https://www.ccadb.org/resources). Questions related to an individual Root Store Operator Policy SHOULD be directed to the operator of that policy using the contact address designated for that policy.

Reviewer 3: ```suggestion
Root Store Operators MAY have additional CCADB-related requirements defined in their own Root Store policies. Those policies MAY strengthen the requirements described in this policy. Additional information on the policies belonging to the individual Root Store Operators participating in the CCADB can be found [here](https://www.ccadb.org/resources). Questions related to an individual Root Store Operator Policy SHOULD be directed to the operator of that policy using the contact address designated for that policy.
```

To promote simplicity and clarity, all CA policy documents SHOULD be:
- focused on one specific PKI use case (e.g., TLS server authentication, TLS client authentication, S/MIME, Code Signing, etc.), rather than combining multiple use cases into a single document or set of documents.
- comprehensive and consolidated, whenever possible, such that there are not multiple sets of similar yet slightly different policy and practice statements supporting the same PKI use case.
- available in Markdown or AsciiDoc.

CA Owners MUST strictly adhere to their policy document(s) as disclosed within the CCADB (and not marked as “Superseded”). This extends to all policy documents the CA Owner publishes in relation to its CAs included in a Root Store, such as TSPS documents.

Before corresponding policy changes are put into practice, CA Owners:
- MUST minimally ensure the updated versions of a CA's policy document(s) are uploaded to their own publicly accessible repository, and

> - SHOULD ensure the updated versions of a CA's policy document(s) are submitted to the CCADB within 7 calendar days of the policy document's effective date, but MUST do so within 14 calendar days.

Reviewer 3: The grammar and causality here feel weird: "_Before_ putting changes into practice, CA owners SHOULD ensure updated docs are submitted _within 7 days of the effective date_".

Just from a grammar perspective, we have two temporal adverbial clauses ("before putting changes into practice" and "within 7 days of the effective date") which both apply to the same verb phrase ("owners SHOULD ensure updated docs are submitted") with equal precedence.

From an actual practice perspective, a CA must be in compliance with their CP/CPS at all times, so whenever that document becomes effective, they must make the corresponding policy changes at the same time. So how are they writing a document with an effective date, then waiting up to 7 days, then uploading the document, _then_ changing their actual practices?

Reviewer 2: Agree with your actual practice perspective. 

```suggestion
CA Owners MUST ensure that updated versions of a CA's policy document(s) are uploaded to their own publicly accessible repository before the corresponding policy changes are put into practice. Corresponding policy changes are considered to be put into practice on the effective date specified in the policy document(s), unless individual requirements are explicitly future dated.
Separately, CA Owners SHOULD ensure these updated policy document(s) are submitted to the CCADB within 7 calendar days of the policy document's effective date, and MUST ensure they are submitted within 14 calendar days of that effective date.
```

CA Owners MUST:
- submit an [Add/Update Root Request Case](https://www.ccadb.org/cas/updates) to add or update policy documents for root CA certificates stored in the CCADB.
- add or update  policy documents for subordinate CA certificates directly on the record in CCADB (unless the exceptions stated below apply).

Unless the exceptions below apply, CA policy disclosures for each certificate disclosed to the CCADB MUST be consistent with the following guidance:

> If the parent certificate record discloses a CP:
> - A) It MUST disclose a CPS
> - B) It MUST NOT disclose a CP/CPS
> - C) All child certificate records MUST either:
>      - (1) disclose a combined CP/CPS; or
>      - (2) disclose (a) a CP (either select "Same as Parent" checkbox or a different CP) and (b) a CPS (either select "Same as Parent" checkbox or a different CPS).
> 
> If the parent certificate record discloses a CPS:
> - A) It MUST disclose a CP
> - B) It MUST NOT disclose a CP/CPS
> - C) All child certificate records MUST either:
>      - (1) disclose a combined CP/CPS; or
>      - (2) disclose (a) a CPS (either select "Same as Parent" checkbox or a different CPS) and (b) a CP (either select "Same as Parent" checkbox or a different CP).

Reviewer 3: I think these could be combined into a single paragraph:
```suggestion
If the parent certificate record discloses a CP or CPS:
- A) It MUST disclose both a CP and a CPS
- B) It MUST NOT disclose a CP/CPS
- C) All child certificate records MUST either:
     - (1) disclose a combined CP/CPS; or
     - (2) disclose (a) a CP (either select "Same as Parent" checkbox or a different CP) and (b) a CPS (either select "Same as Parent" checkbox or a different CPS).
```

- add or update audit statements for subordinate CA certificates directly on the record in the CCADB (unless the exceptions stated above apply).

If an audit or audit statement disclosed to the CCADB does not meet the requirements of this policy, then a Root Store Operator MAY require that the CA Owner obtain a new audit, at the CA Owner's expense, for the period of time in question. Additionally, depending on the nature of concerns with the audit, a Root Store Operator MAY require that the CA Owner obtain such an audit from a new auditor.

The presence of qualifications in an audit statement is not, by itself, generally considered a reason to remove a CA Owner from a Root Store. The purpose of audits is to honestly and thoroughly assess a CA Owner’s compliance with requirements which are necessary to assure a secure and stable ecosystem. Audit findings, including qualifications, can help to identify opportunities for improvement, whether for individual CAR Owners or for the wider industry as a whole.

### 5.1 Scope of Certificates Covered by an Audit

> CA certificates MUST appear in the Issuer’s annual audit statement, beginning with the statement corresponding to the audit period when the certificate was issued. Those certificates MUST continue to appear in audit statements until having appeared on at least one audit statement following revocation or key destruction (observed and reported by a Qualified Auditor).

Reviewer 3: This implies that revocation is sufficient to stop auditing a subordinate CA, but expiration is not. Is that the intent?

Reviewer 2: The implication is unintentional.

```suggestion
CA certificates MUST appear in the Issuer’s annual audit statement, beginning with the statement corresponding to the audit period when the certificate was issued. Those certificates MUST continue to appear in audit statements until having appeared on at least one audit statement following revocation, expiration, or key destruction (observed and reported by a Qualified Auditor).
```

### 6.3 Cross-certification across PKI Hierarchies

Some Root Store Operators participating in the CCADB maintain policy requirements such that hierarchies included in the corresponding root store(s) are dedicated to a specific PKI use case, for example only supporting TLS server authentication. 

In support of ubiquity use-cases, CA Owners sometimes issue cross-certificates from non-dedicated hierarchy Root CAs (i.e., “multi-purpose hierarchies”) to dedicated-use hierarchy Root CAs. 

Beginning June 15, 2025, when new cross-certificates are issued across PKI hierarchies and the Subject CA’s public key (a) exists in or (b) might exist in a publicly-trusted self-signed Root CA certificate whose hierarchy should be considered dedicated to a specific PKI use case, the following EKU values MUST exist:

| If the subject hierarchy is dedicated to… | EKU Presence? | EKU OID Permitted? |
|---|---|---|
| TLS server authentication | MUST | Only 1.3.6.1.5.5.7.3.1                       |
| TLS client authentication | MUST | Only 1.3.6.1.5.5.7.3.2                       |

> | TLS (generic)             | MUST | Only 1.3.6.1.5.5.7.3.1 and 1.3.6.1.5.5.7.3.2 |

Reviewer: This seems to allow for TLS server auth and TLS client auth still combined. Should the same allowance be specified for S/MIME as well, since the SBRs allow both?

* How the audit team and team members are bound by law, regulation or professional standards to render an independent assessment of the CA (e.g., https://pub.aicpa.org/codeofconduct/Ethics.aspx# 0.300.050 Objectivity and Independence; CPA Canada, Rule 204; or Annex A of ETSI EN 319 403/403-1, respectively); and
* Whether the audit team relied on any third-party specialists or affiliate audit firms, and if so, their names and where they performed services.

## 6. Additional Required Practices

### 6.1 Disclosing and Responding to Incidents

> The CCADB [Incident Reporting Guidelines (IRGs)](https://www.ccadb.org/cas/incident-report) intend to present a common format for publicly disclosing incidents. CA Owners who are required to publicly disclose incidents MUST adhere to the CCADB IRGs. Those CA Owners MUST update the incidents and respond to questions in accordance with the IRGs until closure.

Reviewer 3: I think it would be reasonable to read the last sentence here as saying that, if the IRGs are updated in the middle of the incident process (i.e. after the report has been filed, but before it has been closed), then the report itself needs to be updated to be in accordance with the new IRGs. I don't think that's the intent (I think this just means that you need to update the bug every 7ish days), so maybe this should be rephrased a little.

Reviewer 2: Agree, they would not need to attempt to update past content, but we would expect any subsequent updates SHOULD align with the new IRGs, but MUST align with the new IRGs if superseded by more than 30 days. We could rephrase the statement to:

```suggestion
The CCADB [Incident Reporting Guidelines (IRGs)](https://www.ccadb.org/cas/incident-report) intend to present a common format for publicly disclosing incidents. CA Owners who are required to publicly disclose incidents:
- SHOULD always use the latest available version of the CCADB IRG templates, and
- MUST NOT use a version of an IRG template that has been superseded by more than 30 calendar days before the corresponding report (i.e., preliminary, full, or closure) is posted.
```
